### ISSUE TITLE
fail to get meaningful response using pytorch-seq2seq for chatbot

### ISSUE THREAD

**Issue by DataTerminatorX** [IBM/pytorch-seq2seq#66]
I'm using pytorch-seq2seq for chatbot. I used two dataset [ubuntu](https://drive.google.com/file/d/0B_bZck-ksdkpVEtVc1R6Y01HMWM/view) and [twitter](https://github.com/suriyadeepan/practical_seq2seq/blob/master/datasets/twitter/pull_raw_data). I 've formatted the datasets, modified data path in "example.py" and tuned some hyper-parameters(e.g. `hidden_size` `batch_size` `epoches`). 

While I fail to get meaningful response after model finished training. When I typed in some sentences like `hello`  `how are you`, it often gave me ['EOS'] or ['i', 'i', 'EOS']. Is there any suggestion to handle this issue?

**Comment by kylegao91** [comment]
@DataTerminatorX Thank you for making the feedback!  Can you post the training logs?  Did the loss converge?  Do you have a develop set to see if the model's overfitted?

**Comment by DataTerminatorX** [comment]
@kylegao91  Thx for reply. Here is the training log. Seems that the model trained well in term of *perplexity*

```
2017-09-05 14:29:29,968 seq2seq.trainer.supervised_trainer INFO     Time elapsed: 11m 30s, Progress: 100%, Train Perplexity: 608.8577
2017-09-05 14:29:31,554 seq2seq.trainer.supervised_trainer INFO     Finished epoch 10, Dev Perplexity: 591.7412
Type in a source sequence:hi
['EOS']
Type in a source sequence:hello
['EOS']
Type in a source sequence:how are you
['EOS']
```

The hyper parameters are
```
hidden_size = 64
batch_size = 256
num_epoch = 10
```

For fast training and validation, the model is trained on a sampled dataset. I attached them here
[tiwtter_dataTrain_sample.txt](https://github.com/IBM/pytorch-seq2seq/files/1276334/tiwtter_dataTrain_sample.txt)
[tiwtter_dataDev_sample.txt](https://github.com/IBM/pytorch-seq2seq/files/1276336/tiwtter_dataDev_sample.txt)

 here.

**Comment by DataTerminatorX** [comment]
BTW, I found this `pytorch-seq2seq` project well organized from programming design view. Hope I could contribute to this project once this issue is solved. We could add a chatbot feature to it.

**Comment by kylegao91** [comment]
@DataTerminatorX Thanks for expressing the interests.  I will look into the issue and let you know.

**Comment by kylegao91** [comment]
I run the sample data with hidden_size=256, and I found that the log you show is far from convergence.  See the log and some results (still don't make much sense but better than empty) I got after 20 epoches below:

```
2017-09-05 17:14:03,380 seq2seq.trainer.supervised_trainer INFO     Progress: 95%, Train Perplexity: 9.3085
2017-09-05 17:14:56,364 seq2seq.trainer.supervised_trainer INFO     Progress: 95%, Train Perplexity: 7.7437
2017-09-05 17:15:49,935 seq2seq.trainer.supervised_trainer INFO     Progress: 96%, Train Perplexity: 8.1597
2017-09-05 17:16:42,897 seq2seq.trainer.supervised_trainer INFO     Progress: 97%, Train Perplexity: 8.1890
2017-09-05 17:17:35,497 seq2seq.trainer.supervised_trainer INFO     Progress: 98%, Train Perplexity: 7.7902
2017-09-05 17:18:28,137 seq2seq.trainer.supervised_trainer INFO     Progress: 99%, Train Perplexity: 8.5640
2017-09-05 17:19:21,026 seq2seq.trainer.supervised_trainer INFO     Progress: 99%, Train Perplexity: 8.4239
2017-09-05 17:20:00,278 seq2seq.trainer.supervised_trainer INFO     Finished epoch 20: Train Perplexity: 8.1201, Dev Perplexity: 6.3091

Type in a source sequence:how are you
[u"you're", u'me', u'ever', u'amp', u'what', u'are', u'you', u'to', '<eos>']
Type in a source sequence:i'm dead not looking forward to thi
[u"shouldn't", u'has', u'the', u'natural', u'answer', u'to', u'hashtag', u'hashtag', u'hashtag', u'to', '<eos>']
Type in a source sequence:i m waiting for to wake up p
[u'then', u'some', u'some', u'sf', u'hipsters', u'would', u'rather', u'get', u'rid', u'of', u'and', '<eos>']
Type in a source sequence:and then there's these folks
[u'the', u'good', u'place', '<eos>']
Type in a source sequence:i'm good how are you
[u'good', u'good', u'too', u'what', u'are', u'you', '<eos>']
Type in a source sequence:how are you
[u"you're", u'me', u'ever', u'amp', u'what', u'are', u'you', u'to', '<eos>']
Type in a source sequence:hellow
[u"it's", u"we're", u'all', u'i', u"can't", '<eos>']
Type in a source sequence:hello
[u'girl', u'girl', u'maria', u'here', u'at', u'here', u'mocking', u'mocking', u'mocking', '<eos>']
Type in a source sequence:my life story
[u'this', u'is', u'why', u'is', u"we're", '<eos>']
```

**Comment by kylegao91** [comment]
@DataTerminatorX I will close this issue.  Let us know if you have more questions.
